projects/manager/src/app/datasets/dataset-editor/dataset-editor.component.ts

```diff
@@ -253,6 +321,41 @@ export class DatasetEditorComponent implements OnChanges {
             description: new FormControl('description', {
                 nonNullable: true,
             }),
+            tags: new FormControl<string[]>([], {nonNullable: true, validators: [duplicateTagValidator()]}),
+            newTag: new FormControl('', {nonNullable: true, validators: [tagValidator()]}),
         });
     }
 }
+
+export const duplicateTagValidator =
+    (): ValidatorFn =>
+    (control: AbstractControl): ValidationErrors | null => {
+        const tags = control.value as string[];
+
+        if (!tags) {
+            return null;
+        }
+
+        const duplicates = tags.some((value, index) => tags.indexOf(value) !== index);
+
+        return duplicates ? {duplicate: true} : null;
+    };
```

Comment:
Is a duplicate validator maybe something that we could put under utils/validators?

Reply:
83aaed8457b2ed135871dae675ab4dc294fca7a9